builtin/providers/vsphere/resource_vsphere_virtual_machine_test.go

```diff
@@ -750,6 +750,365 @@ func TestAccVSphereVirtualMachine_ipv4Andipv6(t *testing.T) {
+					resource.TestCheckResourceAttr(
+						"vsphere_virtual_machine.linked_current_snapshot_with_extra_disk", "network_interface.#", "1"),
+					resource.TestCheckResourceAttr(
+						"vsphere_virtual_machine.linked_current_snapshot_with_extra_disk", "network_interface.0.label", label),
+				),
+			},
+		},
+	})
+}
+
+// TestAccVSphereVirtualMachine_templateAsString tests if it works to create a clone by using the deprecated template attribute, instead of the new clone attribute.
+func TestAccVSphereVirtualMachine_templateAsString(t *testing.T) {
+	var vm virtualMachine
+	var locationOpt string
+	var datastoreOpt string
+
+	if v := os.Getenv("VSPHERE_DATACENTER"); v != "" {
+		locationOpt += fmt.Sprintf("    datacenter = \"%s\"\n", v)
+	}
+	if v := os.Getenv("VSPHERE_CLUSTER"); v != "" {
+		locationOpt += fmt.Sprintf("    cluster = \"%s\"\n", v)
+	}
+	if v := os.Getenv("VSPHERE_RESOURCE_POOL"); v != "" {
+		locationOpt += fmt.Sprintf("    resource_pool = \"%s\"\n", v)
+	}
+	if v := os.Getenv("VSPHERE_DATASTORE"); v != "" {
+		datastoreOpt = fmt.Sprintf("        datastore = \"%s\"\n", v)
+	}
+	cloneSource := os.Getenv("VSPHERE_CLONE_SOURCE")
+	label := os.Getenv("VSPHERE_NETWORK_LABEL_DHCP")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckVSphereVirtualMachineDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: fmt.Sprintf(
+					testAccCheckVSphereVirtualMachineConfig_templateAsString,
+					locationOpt,
+					label,
+					datastoreOpt,
+					cloneSource,
+				),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVSphereVirtualMachineExists("vsphere_virtual_machine.template_as_string", &vm),
+					resource.TestCheckResourceAttr(
+						"vsphere_virtual_machine.template_as_string", "name", "terraform-test-template-as-string"),
+					resource.TestCheckResourceAttr(
+						"vsphere_virtual_machine.template_as_string", "vcpu", "2"),
+					resource.TestCheckResourceAttr(
+						"vsphere_virtual_machine.template_as_string", "memory", "4096"),
+					resource.TestCheckResourceAttr(
+						"vsphere_virtual_machine.template_as_string", "disk.#", "1"),
+					resource.TestCheckResourceAttr(
+						"vsphere_virtual_machine.template_as_string", "disk.0.template", cloneSource),
+					resource.TestCheckResourceAttr(
+						"vsphere_virtual_machine.template_as_string", "disk.0.type", "thin"),
+					resource.TestCheckResourceAttr(
+						"vsphere_virtual_machine.template_as_string", "network_interface.#", "1"),
+					resource.TestCheckResourceAttr(
+						"vsphere_virtual_machine.template_as_string", "network_interface.0.label", label),
+				),
+			},
+		},
+	})
+}
+
+// TestAccVSphereVirtualMachine_deprecatedLinkedClone tests if it works to create a clone by using the deprecated template and linked cloning attributes.
+func TestAccVSphereVirtualMachine_deprecatedLinkedClone(t *testing.T) {
```

Comment:
NICE TESTING ... You get a free cookie!!!  

Would it be worth to write a brief blurb about how to setup the tests?  The tests are awesome, but not trivial to get setup.